lib/membrane_opus/encoder.ex

```diff
@@ -69,14 +69,20 @@ defmodule Membrane.Opus.Encoder do
     end
   end
 
+  @impl true
+  def handle_prepared_to_playing(_ctx, %{input_caps: caps} = state) when not is_nil(caps) do
+    output_caps = %Opus{channels: caps.channels}
+    {{:ok, caps: {:output, output_caps}}, state}
+  end
+
   @impl true
   def handle_caps(:input, caps, _ctx, %{input_caps: nil} = state) do
     output_caps = %Opus{channels: caps.channels}
     {{:ok, caps: {:output, output_caps}}, %{state | input_caps: caps}}
   end
 
   @impl true
-  def handle_caps(:input, caps, _ctx, %{input_caps: caps} = state) do
+  def handle_caps(:input, _caps, _ctx, %{input_caps: _input_caps} = state) do
     {:ok, state}
   end
```

Comment:
Do we ever reach the next `handle_caps` clause?

Reply:
I have just changed it. Indeed, we need to check if the caps sent are the caps hold in the state.